docs/DeveloperGuide.md

```diff
@@ -237,70 +237,403 @@ _{Explain here how the data archiving feature will be implemented}_
+
+    * 4a1. Productiv cancels the deletion process.
+    
+      Use case ends
+
+#### MEETING
+**Use case: `UC30 - Add a meeting`**
+
+**Precondition(s):**
+* **`User is in the meeting mode`**
+
+**MSS**
+
+1. User adds meeting
+    Use case ends
+    
+**Extensions**
+
+* 1a. Invalid input
+    
+    * 1a1. Productiv shows an error message
+      
+      Use case ends2
+      
+**Use case: `UC31 - Edit a meeting`**
+
+**Precondition(s):**
+* **`User is in the meeting mode`** 
+* **`Meeting to edit exists`**
+
+**MSS**
+
+1. User requests to list meetings
+1. Productiv shows list of meetings
+1. User edits a specific meeting in the list
+1. Productiv edits meeting details
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+  
+  * Use case ends
+     
 * 3a. The given index is invalid.
+    
+    * 3a1. Productiv shows an error message.
+    
+      Use case resumes at step 2
+      
+* 4a. The given input is invalid
+    
+    * 4a1 Productiv shows an error message
+      
+      Use case resumes at step 2
+      
+**Use case: `UC32 - Delete a meeting`**
+
+**Precondition(s):**
+* **`User is in the meeting mode`** 
+* **`Meeting to delete exists`**
+
+**Guarantee(s):** 
+* **`Deleted meeting will not have its data in Productiv`**
+* **`Deleted meeting cannot be retrieved back`**
+
+**MSS**
+
+1. User requests to list meetings
+1. Productiv shows list of meetings
+1. User requests to delete a specific meeting in the list
+1. Productiv deletes the meeting
+   User case ends.
 
-    * 3a1. AddressBook shows an error message.
+    Use case ends.
 
-      Use case resumes at step 2.
+**Extensions**
 
+* 2a. The list is empty.
+  
+  * Use case ends
+     
+* 3a. The given index is invalid.
+    
+    * 3a1. Productiv shows an error message.
+    
+      Use case resumes at step 2
+      
 *{More to be added}*
+      
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+1. Should be able to hold up to 1000 contacts without any negative effects.
+1. Should be able to hold up to 1000 deliverables without any negative effects.
+1. Should be able to schedule meetings up to 2 years in advance.
+1. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 *{More to be added}*
 
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
 * **Private contact detail**: A contact detail that is not meant to be shared with others
```

Comment:
Should we leave this keyphrase out since I don't think we use it anywhere?

Reply:
Noted, will remove accordingly